### PR TITLE
Formatted to PEP8 guidelines

### PR DESCRIPTION
**Pull request by ChristopherBilg**

### CODE REVIEW

File: algorithms.py [hunk 1]
```diff
@@ -4,215 +4,245 @@
...
                 problem.set_state(next_state)
                 attempts = 0
-                
+
             else:
                 attempts += 1
-        
-        # Update best state and best fitness        
+
+                # Update best state and best fitness
```
Comment (42B) [inline algorithms.py]: Is this indentation correct?

File: fitness.py [hunk 2]
```diff
@@ -4,397 +4,408 @@
...
-    for i in range(len(x)):
-        if x[i] == b:
-            head += 1
+    for i in range(len(_x)):
+        if _x[i] == _b:
+            _head += 1
         else:
             break
 
     return head
```
Comment (42B) [inline fitness.py]: Would this have to be `return _head` too?

File: fitness.py [hunk 3]
```diff
@@ -4,397 +4,408 @@
...
-    for i in range(len(x)):
-        if x[len(x) - i - 1] == b:
-            tail += 1
+    for i in range(len(_x)):
+        if _x[len(_x) - i - 1] == _b:
+            _tail += 1
         else:
             break
 
     return tail
```
Comment (42B) [inline fitness.py]: `return _tail`?

File: fitness.py [hunk 4]
```diff
@@ -4,397 +4,408 @@
...
         else:
-            if run > max_run:
-                max_run = run
-                
+            if run > _max:
+                _max = run
+
             run = 0
 
     return max_run
```
Comment (42B) [inline fitness.py]: `return _max`?

File: fitness.py [hunk 5]
```diff
@@ -4,397 +4,408 @@
...
-        
+
         # Determine fitness
-        if (np.max(row_sums) == 1 and np.min(row_sums) == 1) and \
-            (np.max(col_sums) == 1 and np.min(col_sums) == 1) and \
-            diag_sum == 0:
-                
+        if(np.max(row_sums) == 1 and np.min(row_sums) == 1) \
+          and (np.max(col_sums) == 1 and np.min(col_sums) == 1) \
+          and diag_sum == 0:
```
Comment (42B) [inline fitness.py]: Is this indentation really PEP8 compliant?

**Comment by ChristopherBilg** [comment]
These variable are innately private. The underscore is there to enforce PEP8 compliance as one letter variable names are not PEP8 compliant.

**Comment by 42B** [comment]
@ChristopherBilg 

...and what about the bare assert statements?

**Comment by ChristopherBilg** [comment]
The assert statements shouldn't be bare. I'll correct that. I was transitioning from python 2 to 3 during the PEP8 format checking.

**Pull request merged**